### ISSUE TITLE
[FP] Illusion of choice: Potential change of behaviour

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1212]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate),2.0.13 |

Current behaviour
--

For the following code:
```php
<?php

class a{
    public function get()
    {
        $b = "";
        if ($b != null)
        {
            return $b;
        }
        else
        {
            return null;
        }
    }
}
$a = new a();
var_dump($a->get());
```

The method get() can't return an empty string. The weak comparison with null will catch empty strings, 0, empty array... and just return null
However, the inspection suggest to replace by return $b who will potentially allow those values.

For a system with strict type checks this may be an issue.

This can be fixed by checking if the comparison with null is done with ===, but it reduce the scope of the inspection by a lot...

Maybe just adding a warning on the inspection would be sufficient?
